src/frontend/src/tests/instance_test.rs

```diff
@@ -512,6 +512,179 @@ async fn test_execute_external_create_without_ts_type(instance: Arc<dyn MockInst
+| c     | Boolean | YES  |         | FIELD         |
+| d     | String  | YES  |         | FIELD         |
+| e     | Int64   | YES  |         | FIELD         |
+| f     | String  | YES  |         | FIELD         |
+| g     | String  | YES  |         | FIELD         |
++-------+---------+------+---------+---------------+";
+    check_output_stream(output, expect).await;
+
+    let output = execute_sql(&instance, &format!("select * from {table_name};")).await;
+    let expect = "\
++-----------------+------+-------+------+------------+----------------+-------------------------+
+| a               | b    | c     | d    | e          | f              | g                       |
++-----------------+------+-------+------+------------+----------------+-------------------------+
+| 1               | 2.0  | false | 4    | 1681319393 | 1.02           | 2012-04-23T18:25:43.511 |
+| -10             | -3.5 | true  | 4    | 1681356393 | -0.3           | 2016-04-23T18:25:43.511 |
+| 2               | 0.6  | false | text | 1681329393 | 1377.223       |                         |
+| 1               | 2.0  | false | 4    |            | 1337.009       |                         |
+| 7               | -3.5 | true  | 4    |            | 1              |                         |
+| 1               | 0.6  | false | text |            | 1338           | 2018-10-23T18:33:16.481 |
+| 1               | 2.0  | false | 4    |            | 12345829100000 |                         |
+| 5               | -3.5 | true  | 4    |            | 99999999.99    |                         |
+| 1               | 0.6  | false | text |            | 1              |                         |
+| 1               | 2.0  | false | 4    |            | 1              |                         |
+| 1               | -3.5 | true  | 4    |            | 1              |                         |
+| 100000000000000 | 0.6  | false | text |            | 1              |                         |
++-----------------+------+-------+------+------------+----------------+-------------------------+";
+    check_output_stream(output, expect).await;
+}
+
+#[apply(both_instances_cases)]
+async fn test_execute_query_external_table_json_with_schame(instance: Arc<dyn MockInstance>) {
+    let instance = instance.frontend();
+    let format = "json";
+    let location = get_data_dir("../../tests/data/json/various_type.json")
+        .canonicalize()
+        .unwrap()
+        .display()
+        .to_string();
+
+    let table_name = "various_type_json_with_schema";
+
+    let output = execute_sql(
+        &instance,
+        &format!(
+            r#"CREATE EXTERNAL TABLE {table_name} (
+                a BIGINT NULL,
+                b DOUBLE NULL,
+                c BOOLEAN NULL,
+                d STRING NULL,
+                e TIMESTAMP(0) NULL,
+                f DOUBLE NULL,
+                g TIMESTAMP(0) NULL,
+              ) WITH (location='{location}', format='{format}');"#,
```

Comment:
I checked the `arrow-json`; currently, it cannot infer the `timestamp` and `date` types. We can create a PR for this feature in the future.